### PR TITLE
use Enum for resource type

### PR DESCRIPTION
**Pull request by RaphaelVogel**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator

```

### CODE REVIEW

File: whd/dispatcher.py [hunk 1]
```diff
@@ -220,9 +223,9 @@ def _set_pr_labels(self, pr_event, resources):
 
     def _matching_resources(self, concourse_api, event):
         if isinstance(event, PushEvent):
-            resource_type = 'git'
+            resource_type = ResourceType.GIT.value
```
Comment (AndreasBurger) [inline whd/dispatcher.py:226]:
```suggestion
            resource_type = ResourceType.GIT
```

File: whd/dispatcher.py [hunk 2]
```diff
@@ -220,9 +223,9 @@ def _set_pr_labels(self, pr_event, resources):
...
         elif isinstance(event, PullRequestEvent):
-            resource_type = 'pull-request'
+            resource_type = ResourceType.PULL_REQUEST.value
```
Comment (AndreasBurger) [inline whd/dispatcher.py:228]:
```suggestion
            resource_type = ResourceType.PULL_REQUEST
```

File: concourse/client/util.py [hunk 3]
```diff
@@ -32,7 +33,7 @@ def determine_pr_resource_versions(
 
     pr_resources = concourse_api.pipeline_resources(
         pipeline_names=concourse_api.pipelines(),
-        resource_type='pull-request',
+        resource_type=ResourceType.PULL_REQUEST.value,
```
Comment (AndreasBurger) [inline concourse/client/util.py:36]:
```suggestion
        resource_type=ResourceType.PULL_REQUEST,
```

File: concourse/model/resources.py [hunk 4]
```diff
@@ -228,9 +231,9 @@ def __init__(
 
         # todo: handle "qualifier"
         if is_pull_request:
-            type_name = 'pull-request'
+            type_name = ResourceType.PULL_REQUEST.value
         else:
-            type_name = 'git'
+            type_name = ResourceType.GIT.value
```
Comment (AndreasBurger) [inline concourse/model/resources.py:237]: Could pbly also be replaced with the type and updating the ResourceIdentifier. Not sure, and either way I'm fine with doing this at a later point.

File: concourse/resources/variants.mako [hunk 5]
```diff
@@ -1,4 +1,7 @@
 <%def name="render_repositories(pipeline_definition, cfg_set)">
+<%
+from concourse.client.model import ResourceType
+%>
 <%namespace file="/resources/github.mako" import="*"/>
 % for repo_cfg in pipeline_definition._resource_registry.resources(type_name='git'):
```
Comment (AndreasBurger) [inline concourse/resources/variants.mako:6]: You've forgotten to replace it here:
```suggestion
% for repo_cfg in pipeline_definition._resource_registry.resources(type_name=ResourceType.GIT.value):
```

Comment (RaphaelVogel) [reply concourse/resources/variants.mako]: done

**Pull request merged**